src/store.c

```diff
@@ -437,6 +439,23 @@ static int _must_use_ _nonnull_ cs_content_add(struct clip_store *cs,
         }
 
         return 0;
+    } else if (dupe && deduplicate) {
+        _drop_(cs_unref) struct ref_guard guard = cs_ref(cs);
+        if (guard.status < 0) {
+            return guard.status;
+        }
+
+        for (int i = 0; i < (int)cs->local_nr_snips; ++i) {
+            if (cs->snips[i].hash == hash) {
+                // move cs->snips[i] to the end of the array
+                struct cs_snip tmp = cs->snips[i];
```

Comment:
Also, this logic is in the wrong place. This function is for adding content, but this adjusts the snip cache. So this should go into a snip related function, not here.